### PR TITLE
webpack v4: Add infrastructureLogging option

### PR DESCRIPTION
**Pull request by andersk**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://v4.webpack.js.org/configuration/other-options/#infrastructurelogging>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by andersk** [comment]
CI failure is unrelated: microsoft/TypeScript#45804.

**Review by ab455 — APPROVED** [review]

**Review by peterblazejewicz — APPROVED** [review]

**Comment by andersk** [comment]
(Rebased with no changes for passing CI.)

**Review by peterblazejewicz — APPROVED** [review]

**Comment by andersk** [comment]
Ready to merge

File: types/webpack/v4/test/index.ts [hunk 1]
```diff
@@ -1249,3 +1249,14 @@ const config2: webpack.MultiConfigurationFactory = (env) => {
     env; // $ExpectType string | Record<string, string | number | boolean> | undefined
     return [];
```
Comment (hsnalsafy) [inline types/webpack/v4/test/index.ts:1250]: 1250

**Review by ab455 — APPROVED** [review]

**Pull request merged**